### ISSUE TITLE
@prisma/client/edge + Cloudflare Worker / wrangler = Could not resolve "os"

### ISSUE THREAD

**Issue by osejrafael** [prisma/prisma#19174]
### Bug description

It was running just fine. I updated prisma and reinstalled dependencies then did `npx prisma generate --data-proxy`. Generates the client, but right after starting the server with `wrangler dev src/index.ts` I get 
<img width="687" alt="Screenshot 2023-05-10 at 7 11 57 AM" src="https://github.com/prisma/prisma/assets/13815017/41170c3b-1b39-4463-96a3-1af852228b97">

### How to reproduce

1. Run a basic Cloudflare worker `wrangler.toml`
```toml
name = "api"
compatibility_date = "2023-04-19"
```
2. Generate prisma with data proxy `npx prisma generate --data-proxy`
3. Setup/connect prisma.
4. Start wrangler `wrangler dev src/index.ts`
5. Get error

### Expected behavior

Server starts and Prisma edge client works.

### Prisma information

```javascript
import { PrismaClient } from "@prisma/client/edge";

export default function prisma(url: string) {
  const client = new PrismaClient({
    datasources: {
      db: {
        url,
      },
    },
  });

  return client;
}

export type ExtendedPrismaClient = ReturnType<typeof prisma>;

```

```prisma
generator client {
  provider = "prisma-client-js"
  previewFeatures = ["clientExtensions", "jsonProtocol"]
  // ALSO TESTED WITH NO PREVIEW FEATURES
}

datasource db {
  provider = "mysql"
  url = env("DATABASE_URL")
  relationMode = "prisma"
}
```

### Environment & setup

- OS: macOS Apple M2 Pro
- Database: PlanetScale MySQL
- Node: 18.16

### Prisma Version

```
4.14
```

**Comment by osejrafael** [comment]
Update: It works when I manually set `Rp={}` instead of `Rp=require("os")` in `edge.js`

Before:
<img width="186" alt="Screenshot 2023-05-10 at 7 47 13 AM" src="https://github.com/prisma/prisma/assets/13815017/0d9baa34-9883-4e14-90db-51d02c5337da">

After:
<img width="187" alt="Screenshot 2023-05-10 at 7 47 07 AM" src="https://github.com/prisma/prisma/assets/13815017/29e7fdfc-6dd8-404a-aa80-8b38d41e3810">

**Comment by Jolg42** [comment]
@osejrafael Thanks for this issue! Do you know if this error also happens with a previous version of Prisma, like `4.13.0` or earlier? Maybe this is a recent change 🤔

**Comment by osejrafael** [comment]
@Jolg42 Yes, I tried rolling back to `4.13.0` and I got the same error. Which is weird since that was the only change I made. "Update from `4.13.0` -> `4.14.0` and generate client with `--data-proxy`"

**Comment by alexvuka1** [comment]
I have been using `@prisma/client/edge` on Vercel Edge (using `@astrojs/vercel/edge`) and it has been working fine on version `4.13.0`, but when I tried updating to `4.14.0` I got the same error as @osejrafael

**Comment by oanogin** [comment]
I have the same issue with 4.14.0 :/
Workaround 1: replace `Rp=require("os")` with `Rp={}` in `../node_modules/@prisma/client/runtime/edge.js`
Workaround 2: add `node_compat = true` line to wrangler.toml file

**Comment by Jolg42** [comment]
Thanks all! I'll try a reproduction tomorrow, as this looks like an unexpected regression 👀 

In the meantime, I can recommend staying on `4.13.0` and stay tuned.

**Comment by alexvuka1** [comment]
The file `ExitHooks.ts` is new in `4.14.0` and it imports `os`:

https://github.com/prisma/prisma/blob/18313b3747f5c4ef62408c470d661beb312022a7/packages/client/src/runtime/core/engines/library/ExitHooks.ts#L2

Not sure if it is the root of the problem, but might be useful.

**Comment by Jolg42** [comment]
@alexvuka1 Thanks! I also found that this was the most recent addition, potentially that's the problem here.

**Comment by millsp** [comment]
I can confirm I have seen this in the bundled files. We should add a test that no `require`/`import` exists in our edge runtime. Currently `os` imports are not actually required and we could juts blank it out.

**Comment by Jolg42** [comment]
I got a reproduction in our ecosystem-tests https://github.com/prisma/ecosystem-tests/pull/3508 after updating wrangler
I'll merge my PR for updating our cloudflare worker test, and it will most probably be fixed by https://github.com/prisma/prisma/pull/19186 once merged
We'll come back with a confirmation once all testing went through.

**Comment by Jolg42** [comment]
Note: confirmed fixed on main branch.
Our internal dev version `4.15.0-dev.13` has the fix and [passes the ecosystem-tests](https://github.com/prisma/ecosystem-tests/actions/runs/4950830323/jobs/8855011566#step:8:231)

**Comment by Jolg42** [comment]
Note: we are planning an official patch release later today. Stay tuned 😉
